elytron/src/main/java/org/wildfly/extension/elytron/ModifiableRealmDecorator.java

```diff
@@ -503,31 +537,42 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
 
         private Password createPassword(final OperationContext parentContext, final String principalName, String passwordType, ModelNode passwordNode) throws OperationFailedException, NoSuchAlgorithmException, InvalidKeySpecException {
-            String password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
+            String password;
             final PasswordSpec passwordSpec;
             final String algorithm;
 
             if (passwordType.equals(ElytronDescriptionConstants.BCRYPT)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
                 byte[] salt = SetPasswordHandler.Bcrypt.SALT.resolveModelAttribute(parentContext, passwordNode).asBytes();
                 int iterationCount = SetPasswordHandler.Bcrypt.ITERATION_COUNT.resolveModelAttribute(parentContext, passwordNode).asInt();
                 passwordSpec = new EncryptablePasswordSpec(password.toCharArray(), new IteratedSaltedPasswordAlgorithmSpec(iterationCount, salt));
                 algorithm = SetPasswordHandler.Bcrypt.ALGORITHM.resolveModelAttribute(parentContext, passwordNode).asString();
             } else if (passwordType.equals(ElytronDescriptionConstants.CLEAR)) {
+                password = SetPasswordHandler.Bcrypt.PASSWORD.resolveModelAttribute(parentContext, passwordNode).asString();
```

Comment:
About appending -from, it was verified with @honza889. -from are derived values used in LDAP, OTP will use direct values so attributes without -from are fine here.
About the minimum size, all of the password types aren't optional, we can remove the minimum size in Bcrypt, Clear and SimpleDigest